core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionBuilder.scala

```diff
@@ -8,8 +8,10 @@ class TSqlFunctionBuilder extends FunctionBuilder {
     case "@@CURSOR_STATUS" => FunctionDefinition.notConvertible(0)
     case "@@FETCH_STATUS" => FunctionDefinition.notConvertible(0)
     // The ConversionStrategy is used to rename ISNULL to IFNULL
-    case "ISNULL" => FunctionDefinition.standard(2).withConversionStrategy(TSqlFunctionConverters.FunctionRename)
+    case "ISNULL" => FunctionDefinition.standard(2).withConversionStrategy(TSqlFunctionConverters.Rename)
     case "MODIFY" => FunctionDefinition.xml(1)
+    case "NEXTVALUEFOR" =>
+      FunctionDefinition.standard(1).withConversionStrategy(TSqlFunctionConverters.NextValueFor)
   }
 
```

Comment:
```suggestion
    case "NEXTVALUEFOR" => FunctionDefinition.standard(1).withConversionStrategy(nextValueFor)
  }
  
  private def nextValueFor(irName: String, args: Seq[ir.Expression]): ir.Expression = {
      // Note that this conversion assumes that the CREATE SEQUENCE it references was an increment in ascending order.
      // We may run across instances where this is not the case, and will have to handle that as a special case, perhaps
      // with external procedures or functions in Java/Scala, or even python.
      // For instance a SequenceHandler supplied by the user.
      //
      // Given this, then we use this converter rather than just the simple Rename converter.
      // TODO: Implement external SequenceHandler?
      ir.CallFunction("MONOTONICALLY_INCREASING_ID", List.empty)
    }

```

more about the pattern: conversion strategies are mostly Single Method Interfaces (SMIs), that can be represented as `(String, Seq[ir.Expression]) => ir.Expression` type, so it would improve readability if we don't add extra static final singletons (`object`) just to have one method. In Java8+ it would have been called an anti-pattern. It's definitely anti-pattern in Scala.

Reply:
Moreover, if a trait/abstract class has a single abstract method like this:
```scala
trait Foo{
  def bar(x: X, y: Y): Z
}
``` 

It can be instantiated with simply:
```scala
val foo: Foo = (x, y) => computeSomeZ(x, y)
```